src/validation.cpp

```diff
@@ -2245,7 +2284,7 @@ static void UpdateTip(CTxMemPool& mempool, const CBlockIndex* pindexNew, const C
     }
 
     bilingual_str warning_messages;
-    if (!active_chainstate.IsInitialBlockDownload()) {
+    if (!chainstate.IsInitialBlockDownload()) {
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/21526#discussion_r658163708

> IIUC this log warns node operator about new consensus rules activated through the BIP9 logic. In case of the assumed-utxo-set being dated after either the locked-in or activation, your foreground validation is going to be blinded and your background validation is going to skim over this warning ?

I don't think this is a change because the warning is skipped anyway in IBD. So loading a snapshot doesn't skip this warning because it would be skipped anyway. Loading a snapshot just makes the node more usable while blocks are still being downloaded.

Reply:
I agree it's not a change, my point is more about how assume-utxo is breaking the notion of a sequential IBD and introducing out-of-order validation which might invite us to reconsider the design purpose of such warning.

`IsInitialBlockDownload`'s last check relies on our chain tip being older than DEFAULT_MAX_TIP_AGE (24h) and i think that's a reasonable assumption than hardcoded assumed-utxo will be always older than 24h. So background chainstate is always going to be considered as IBD, and such warning skipped, without giving a _chance_ to flag out during foreground chainstate processing if it did start after the locked-in period.

Though I concede, it's tangential to this PR and belong more to a discussion on how assume-utxo is interacting with softfork deployments.